### PR TITLE
[FIX] github pre-commit action: pin python version to 3.11

### PR DESCRIPTION
**Pull request by em230418**
Recently I had problems with passing pre-commit, since github action started to use Python 3.12

When running pre-commit action on 3.12 you get following errors:

```
autoflake................................................................Failed
- hook id: autoflake
- exit code: 1

Traceback (most recent call last):
  File "/home/runner/.cache/pre-commit/repo7lg_lq3_/py_env-python3/bin/autoflake", line 5, in <module>
    from autoflake import main
  File "/home/runner/.cache/pre-commit/repo7lg_lq3_/py_env-python3/lib/python3.12/site-packages/autoflake.py", line 32, in <module>
    import distutils.sysconfig
ModuleNotFoundError: No module named 'distutils'
```

```
flake8...................................................................Failed
- hook id: flake8
- exit code: 1

Traceback (most recent call last):
  File "/home/runner/.cache/pre-commit/repotxkzga5l/py_env-python3/bin/flake8", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/runner/.cache/pre-commit/repotxkzga5l/py_env-python3/lib/python3.12/site-packages/flake8/main/cli.py", line 22, in main
    app.run(argv)
  File "/home/runner/.cache/pre-commit/repotxkzga5l/py_env-python3/lib/python3.12/site-packages/flake8/main/application.py", line 363, in run
    self._run(argv)
  File "/home/runner/.cache/pre-commit/repotxkzga5l/py_env-python3/lib/python3.12/site-packages/flake8/main/application.py", line 350, in _run
    self.initialize(argv)
  File "/home/runner/.cache/pre-commit/repotxkzga5l/py_env-python3/lib/python3.12/site-packages/flake8/main/application.py", line 330, in initialize
    self.find_plugins(config_finder)
  File "/home/runner/.cache/pre-commit/repotxkzga5l/py_env-python3/lib/python3.12/site-packages/flake8/main/application.py", line 153, in find_plugins
    self.check_plugins = plugin_manager.Checkers(local_plugins.extension)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.cache/pre-commit/repotxkzga5l/py_env-python3/lib/python3.12/site-packages/flake8/plugins/manager.py", line 356, in __init__
    self.manager = PluginManager(
                   ^^^^^^^^^^^^^^
  File "/home/runner/.cache/pre-commit/repotxkzga5l/py_env-python3/lib/python3.12/site-packages/flake8/plugins/manager.py", line 238, in __init__
    self._load_entrypoint_plugins()
  File "/home/runner/.cache/pre-commit/repotxkzga5l/py_env-python3/lib/python3.12/site-packages/flake8/plugins/manager.py", line 254, in _load_entrypoint_plugins
    eps = importlib_metadata.entry_points().get(self.namespace, ())
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'EntryPoints' object has no attribute 'get'
```

Pinning to 3.11 fixes errors above.

### CODE REVIEW

**Comment by legalsylvain** [comment]
Thanks !


Some test are failing. Could you take a look ?

**Comment by em230418** [comment]
> Some test are failing. Could you take a look ?

Python 2.7 issue. Related PR https://github.com/OCA/oca-addons-repo-template/pull/199

File: .github/workflows/lint.yml
Comment (pedrobaeza) [inline .github/workflows/lint.yml:17]: I think this one shouldn't be pinpointed. The problem is in pre-commit.

Comment (em230418) [reply .github/workflows/lint.yml:17]: > I think this one shouldn't be pinpointed. The problem is in pre-commit.

I think otherwise. Github action should not behave differently when suddenly other python version is used.

**Comment by pedrobaeza** [comment]
Being tested in https://github.com/OCA/sale-workflow/pull/2748

**Review by pedrobaeza — APPROVED** [review]
@sbidoul @yajo can you make a new release and maybe launch the massive update for 15/16?

File: .github/workflows/lint.yml
Comment (sbidoul) [reply .github/workflows/lint.yml:17]: I think we must pin the python version for pre-commit, just like we pin it for running tests.
Otherwise we hit this kind of issue every time GitHub upgrades its default python.

My suggestion would be to use the same python version as for the tests.

**Comment by sbidoul** [comment]
> can you make a new release and maybe launch the massive update for 15/16?

I'm not available today, I'll see what I can do this weekend. In the meantime if any board member or core maintainer wants to do the mass update, feel totally free (using oca-copier-update in maintainer-tools).

File: .github/workflows/lint.yml
Comment (pedrobaeza) [reply .github/workflows/lint.yml:17]: Not sure what you want to say here. Can you do a suggestion if still on time?

**Comment by pedrobaeza** [comment]
Thanks, Stéphane. Let's see in @yajo can do at least a release after merging this for manual updates of the needed repos. Other helpful colleagues can be @etobella or @simahawk

**Review by zamberjo — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Version 14 is also affected.

**Review by almumu — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Thank you, Stéphane.

**Pull request merged**
